### PR TITLE
[test] Add some negative tests to tizen_capi

### PR DESCRIPTION
**Pull request by songgot**
There are some case where there is a positive test but no negative test

**Self evaluation:**
1. Build test:   [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Hyunil <hyunil46.park@samsung.com>

### CODE REVIEW

File: test/tizen_capi/unittest_tizen_capi_dataset.cpp [hunk 1]
```diff
@@ -120,9 +120,20 @@ TEST(nntrainer_capi_dataset, create_destroy_06_n) {
 }
 
 /**
- * @brief Neural Network Dataset Create / Destroy Test (positive test )
+ * @brief Neural Network Dataset Create / Destroy Test (negative test)
```
Comment (lhs8928) [inline test/tizen_capi/unittest_tizen_capi_dataset.cpp:123]: Can you delete this test cause I made same one

File: test/unittest/unittest_nntrainer_lr_scheduler.cpp [hunk 2]
```diff
@@ -126,7 +146,27 @@ TEST(lr_exponential, prop_01_n) {
  * @brief test set and get learning rate
  *
  */
-TEST(lr_exponential, prop_02_p) {
+TEST(lr_exponential, prop_02_n) {
```
Comment (lhs8928) [inline test/unittest/unittest_nntrainer_lr_scheduler.cpp:149]: Can you delete this test cause I made same one

File: test/tizen_capi/unittest_tizen_capi_optimizer.cpp [hunk 3]
```diff
@@ -70,7 +70,21 @@ TEST(nntrainer_capi_nnopt, create_delete_04_n) {
...
+ * @brief Neural Network Optimizer Create / Delete Test (negative test)
+ */
+TEST(nntrainer_capi_nnopt, create_delete_05_n) {
+  ml_train_optimizer_h handle;
+  int status;
+
+  status = ml_train_optimizer_create(&handle, ML_TRAIN_OPTIMIZER_TYPE_UNKNOWN);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+
+  status = ml_train_optimizer_destroy(handle);
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi_optimizer.cpp:82]: okay

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 4]
```diff
@@ -509,16 +506,80 @@ TEST(nntrainer_capi_nnmodel, addLayer_04_p) {
   EXPECT_EQ(status, ML_ERROR_NONE);
 }
 
+TEST(nntrainer_capi_nnmodel, addLayer_05_n) {
```
Comment (lhs8928) [inline test/tizen_capi/unittest_tizen_capi.cpp:509]: This test has same effect with addLayer_01_p

File: test/tizen_capi/unittest_tizen_capi_optimizer.cpp [hunk 5]
```diff
@@ -115,6 +129,22 @@ TEST(nntrainer_capi_nnopt, setOptimizer_03_n) {
...
+TEST(nntrainer_capi_nnopt, setOptimizer_04_n) {
+  ml_train_optimizer_h handle;
+  int status;
+  status = ml_train_optimizer_create(&handle, ML_TRAIN_OPTIMIZER_TYPE_UNKNOWN);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+  status = ml_train_optimizer_set_property(
+    handle, "learning_rate=0.0001 | decay_rate=0.96", "decay_steps=1000",
+    "beta1=0.002", "beta2=0.001", "epsilon=1e-7", NULL);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+  status = ml_train_optimizer_destroy(handle);
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi_optimizer.cpp:144]: okay

File: test/tizen_capi/unittest_tizen_capi_optimizer.cpp [hunk 6]
```diff
@@ -115,6 +129,22 @@ TEST(nntrainer_capi_nnopt, setOptimizer_03_n) {
   EXPECT_EQ(status, ML_ERROR_NONE);
 }
 
+/**
+ * @brief Neural Network Optimizer Set Property Test (negative test)
+ */
+TEST(nntrainer_capi_nnopt, setOptimizer_04_n) {
+  ml_train_optimizer_h handle;
+  int status;
+  status = ml_train_optimizer_create(&handle, ML_TRAIN_OPTIMIZER_TYPE_UNKNOWN);
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi_optimizer.cpp:138]: okay

File: test/unittest/unittest_nntrainer_tensor_pool.cpp [hunk 7]
```diff
@@ -483,6 +483,17 @@ TEST(TensorPool, view_is_subset_p) {
...
 }
 
+TEST(TensorPool, view_is_subset_n) {
+  nntrainer::TensorPool pool;
+  // |-------- t1 -------|
+  // |-t2-|
+  //       |-t3-|
+  auto t1 = pool.request("t1", {10}, {0}, max_ls);
+  EXPECT_ANY_THROW(pool.view("t2", "unknown", {3}, {1}, max_ls));
+  EXPECT_ANY_THROW(pool.view("t3", "unknown", {3}, {1}, max_ls, 3));
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/unittest/unittest_nntrainer_tensor_pool.cpp:493]: okay

File: test/tizen_capi/unittest_tizen_capi_dataset.cpp [hunk 8]
```diff
@@ -297,9 +308,23 @@ TEST(nntrainer_capi_dataset, set_dataset_property_02_p) {
...
+ * @brief Neural Network Dataset set Property Test (negative test)
  */
 TEST(nntrainer_capi_dataset, set_dataset_property_03_n) {
+  ml_train_dataset_h dataset = NULL;
+  int status;
+
+  status = ml_train_dataset_set_property(dataset, "buffer_size=100", NULL);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+
+  status = ml_train_dataset_destroy(dataset);
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi_dataset.cpp:320]: okay

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 9]
```diff
@@ -622,7 +683,31 @@ TEST(nntrainer_capi_nnmodel, getLayer_02_p) {
...
  * @brief Neural Network Model Get Layer Test
  */
-TEST(nntrainer_capi_nnmodel, getLayer_03_n) {
+TEST(nntrainer_capi_nnmodel, getLayer_03_p) {
+  int status = ML_ERROR_NONE;
+
+  ml_train_model_h model;
+  ml_train_layer_h get_layer;
+
+  ScopedIni s("getLayer_04_n", {model_base, inputlayer});
```
Comment (lhs8928) [inline test/tizen_capi/unittest_tizen_capi.cpp:692]: getLayer_03_p

File: test/tizen_capi/unittest_tizen_capi_dataset.cpp [hunk 10]
```diff
@@ -490,6 +516,43 @@ TEST(nntrainer_capi_dataset, set_dataset_03_p) {
...
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+
+  status = ml_train_model_destroy(model);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Neural Network Dataset Create / Destroy Test (negative test)
+ */
+TEST(nntrainer_capi_dataset, set_dataset_04_n) {
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi_dataset.cpp:539]: okay

File: test/tizen_capi/unittest_tizen_capi_dataset.cpp (hunk 1 above)
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi_dataset.cpp:123]: okay

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 11]
```diff
@@ -651,7 +776,7 @@ TEST(nntrainer_capi_nnmodel, getLayer_03_n) {
 /**
  * @brief Neural Network Model Get Layer Test
  */
-TEST(nntrainer_capi_nnmodel, getLayer_04_p) {
+TEST(nntrainer_capi_nnmodel, getLayer_06_n) {
```
Comment (lhs8928) [inline test/tizen_capi/unittest_tizen_capi.cpp:779]: This unittest has same effect with getLayer_04_n

File: test/unittest/unittest_nntrainer_lr_scheduler.cpp (hunk 2 above)
Comment (songgot) [reply test/unittest/unittest_nntrainer_lr_scheduler.cpp:149]: okay

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 4 above)
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi.cpp:509]: okay

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 9 above)
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi.cpp:692]: okay

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 11 above)
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi.cpp:779]: okay

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**